File: pset1/mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void) {
    int height;
    do {
        height = get_int("Height: ");
    }
    while (height <= 0 || height >= 9); //Geting Height from user

    for (int i = 0; i < height; i++) { //Generating Rows 
        for (int j = 0; j < height; j++) { //Generating Columns
            if (height - (j + 1) > i) { //Printing Spaces
                printf(" ");
            }
            else {
                printf("#"); //Printing Hashes
            }
        }
        printf("\n"); //Printing New Line at the end of every line
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    while(True):
        height = get_int("Height: ")

        if height > 0 and height <= 8:  # exiting the loop if condition satisfied
            break

    pyramid_maker(height)  # calling a function which makes and return the pattern


def pyramid_maker(h):
    for i in range(0, h):
        for j in range(0, h):
            if(j < (h - (i + 1))):  # condition for printing left pyramid with spaces
                print(' ', end='')
            else:
                print('#', end='')

        print("  ", end='')  # printing space between two columns

        for k in range(0, i+1):
            print('#', end='')  # printing right pyramid without spaces
        print()


main()